### ISSUE TITLE
Environment not cross-platform nor runs out-of-the-box

### ISSUE THREAD

**Issue by anodium** [citra-emu/discord-bot#6]
And for very trivial reasons too, `start.sh` seems to depend on platform-specific `bash` quirks, and it also depends on the npm `forever` module, despite not being mentioned in the `package.json` as a dependency. This confused me for a good 30~40 minutes while trying to write #4. 😠

**Comment by chris062689** [comment]
Forever is designed to keep the service running if it crashes.
https://github.com/foreverjs/foreve

How would you suggest improving the start.sh script? It was really just designed as a launcher for the server, as such it's bash-specific.

**Comment by chris062689** [comment]
How do you propose we resolve this?

**Comment by anodium** [comment]
Two small fixes:

1. Add `forever` as a dependency.
2. Use `forever` from within `server.js`, so as to not require dependency on anything outside of `npm`.

**Comment by chris062689** [comment]
I disagree that the job of the Node.js service is to automatically restart if it crashes, that should be a job of an external tool (such as forever-cli)

**Comment by anodium** [comment]
The thing is that `forever-cli` is written in `node`, so regardless of whether or not it is external, it will still be the same process tree; `forever` at the top level, monitoring a subprocess to see if it dies, and spawning a new one when it does.

It's a tiny change, but it will make managing the entire thing runnable with just `npm run`.

**Comment by chris062689** [comment]
I don't directly consider forever (or any process manager) a dependency of the bot itself.
You should be free to use whichever process manager you like to run the bot.

The startup.sh script only exists as an easy way for me to restart the process on the Citra server.

**Comment by anodium** [comment]
Yeah, which is why I thought to either (1) remove the run script, (2) fully integrate the run script. #10 did the latter.

**Comment by chris062689** [comment]
I'll decouple entirely. And try to create a npm start or npm run entry without any dependency on a process manager or node_env

**Comment by anodium** [comment]
Perfect, I'll close the issue when the PR to decouple is merged. :smile:
